pkg/controllers/cspc-controller/sync.go

```diff
@@ -261,6 +262,25 @@ func (c *Controller) removeCSPCFinalizer(cspc *cstor.CStorPoolCluster) error {
 			"CSPI(s) still exists for CSPC")
 	}
 
+	// If the BD(s) are claimed and the CSPI(s) do not spawn up for some reason,
+	// the pending BDC(s) finalizer don't get removed automatically.
+	// The below code handles cleanup for such cases.
+	bdcList, err := c.GetStoredOpenebsVersionClient().BlockDeviceClaims(cspc.Namespace).List(
+		metav1.ListOptions{
+			LabelSelector: string(types.CStorPoolClusterLabelKey) + "=" + cspc.Name,
+		},
+	)
```

Comment:
Can we handle the error? else it will be the same behavior(BDC will exist) when some error occurs

Reply:
Done